Exiled.CustomItems/API/Features/CustomWeapon.cs

```diff
@@ -59,7 +59,21 @@ public virtual uint ClipSize
         public override float Durability { get; set; }
 
         /// <inheritdoc/>
-        public override void Spawn(Vector3 position) => Spawned.Add(Item.Spawn(Type, ClipSize, position, default, Modifiers.SightType, Modifiers.BarrelType, Modifiers.OtherType));
+        public override void Spawn(Vector3 position, out Pickup pickup)
```

Comment:
We should directly return the pickup instead of using an `out` parameter.
The same applies for every `Spawn` overloads.

On the contrary, using an `out` parameter in `TrySpawn` is perfectly fine.

Reply:
It won't work in our case. They will conflict with deprecated methods that return `void`.

Reply:
^ that. Which was why I told him to make them out parameters, so they can be overloads, otherwise these become breaking changes, and i don't particularly see why they should be. We can always revise this when we introduce other breaking changes, which will also be when we remove all of the [Obsolete] methods, so there would be no more conflict.

Reply:
Yeah. It's like choosing between `static void Method(out int item1, out int item2)` and `static (int, int) Method()`, both are good, but the second one seems to me better, it's more flexible.

Reply:
Ok, if we're forced, then it's fine.